src/platform/Darwin/CHIPPlatformConfig.h

```diff
@@ -62,3 +62,6 @@
 #ifndef CHIP_CONFIG_KVS_PATH
 #define CHIP_CONFIG_KVS_PATH "/tmp/chip_kvs"
 #endif // CHIP_CONFIG_KVS_PATH
+
+// Enable subscription persistence and resumption for CI
+#define CHIP_CONFIG_PERSIST_SUBSCRIPTIONS 1
```

Comment:
Could this be a GN config instead? That way GN build can actually control what cpp and headers to include instead of including empty cpp files to compile.

Reply:
Filed #24564 to track moving this conditional compilation to GN system.